client/src/common/client.ts

```diff
@@ -2767,8 +2767,8 @@ export abstract class BaseLanguageClient {
 		return this._diagnostics;
 	}
 
-	public createDefaultErrorHandler(maxRestartCount: number): ErrorHandler {
-		return new DefaultErrorHandler(this._name, maxRestartCount);
+	public createDefaultErrorHandler(maxRestartCount?: number): ErrorHandler {
+		return new DefaultErrorHandler(this._name, maxRestartCount || 5);
```

Comment:
This is not correct. It will even default to 5 if maxRestartCount is 0 since `||` is coercive. Did you mean maxRestartCount ?? 5

Reply:
Thanks! Ah, right, `??` is what I meant.

Also added a sanity check for invalid value, It should keep the original behavior now.